include/tlapack/lapack/getri.hpp

```diff
@@ -0,0 +1,249 @@
+/// @file getrf.hpp
+/// @author Ali Lotfi, University of Colorado Denver, USA
+//
+// Copyright (c) 2013-2022, University of Colorado Denver. All rights reserved.
+//
+// This file is part of <T>LAPACK.
+// <T>LAPACK is free software: you can redistribute it and/or modify it under
+// the terms of the BSD 3-Clause license. See the accompanying LICENSE file.
+// PA = LU   A^(-1)P^T=U^(-1)L^(-1) --- > U(A^(-1)P^T) L=I
+#ifndef TLAPACK_GETRI_HH
+#define TLAPACK_GETRI_HH
+
+#include "tlapack/base/utils.hpp"
+#include "tlapack.hpp"
+
+namespace tlapack {
+/** getrf computes an LU factorization of a general m-by-n matrix A
+ *  using partial pivoting with row interchanges.
+ *
+ *  The factorization has the form
+ * \[
+ *   A = P L U
+ * \]
+ *  where P is a permutation matrix constructed from our Piv vector, L is lower triangular with unit
+ *  diagonal elements (lower trapezoidal if m > n), and U is upper
+ *  triangular (upper trapezoidal if m < n).
+ *  This is Level 0 version of the algorithm.
+ *
+ * @return  0 if success
+ * @return  i+1 if failed to compute the LU on iteration i
+ *
+ * @param[in,out] A m-by-n complex matrix.
+ *      A=PLU, and to construct L and U, one proceeds as in the following steps
+ *      1. Set matrices L m-by-k, and U k-by-n be to matrices with all zeros, where k=min(m,n)
+ *      2. Set elements on the diagonal of L to 1
+ *      3. below the diagonal of A will be copied to L
+ *      4. On and above the diagonal of A will be copied to U
+ *      
+ * @param[in,out] Piv is a k-by-1 vector where k=min(m,n)
+ * and Piv[i]=j where i<=j<=k-1, which means in the i-th iteration of the algorithm,
+ * the j-th row needs to be swapped with i
+ *
+ * @ingroup group_solve
+ */
+template< class matrix_t, class vector_t >
+int getri( matrix_t& A, vector_t &Piv){
```

Comment:
Piv is not an input. You invert matrix A, and the result is the matrix A inverted. Piv is a local variable of `getri`.

Reply:
Actually, @langou corrected me. getri indeed has an input-only Piv. Please, look on LAPACK.

```
 DGETRI computes the inverse of a matrix using the LU factorization
 computed by DGETRF.
```
 
So, the problem is that getri shouldn't do everything, but just use the precomputed L and U factors to obtain the inverse of A.